app/controllers/api/subcollections/security_groups.rb

```diff
@@ -4,6 +4,32 @@ module SecurityGroups
       def security_groups_query_resource(object)
         object.respond_to?(:security_groups) ? object.security_groups : []
       end
+
+      def security_groups_add_resource(parent, _type, _id, data)
+        security_group = data["security_group"]
+        raise "Cannot add #{security_group} to #{parent.name}" unless parent.respond_to? :add_security_group
```

Comment:
Any reason why this raise is outside of the rescue block? Returning action results for everything plays nicer with bulk actions.

Reply:
was gonna ask, we'd need specs for adding/removing in bulk.